File: CS/15_Document/RemoveBlankPages.cs
using Spire.Pdf;
using Spire.Pdf.Graphics;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace RemoveBlankPages
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new PDF document.
            PdfDocument document = new PdfDocument();

            // Load the file from disk.
            document.LoadFromFile(@"..\..\..\..\..\..\Data\RemoveBlankPages.pdf");

            // Iterate through all pages in reverse order.
            for (int i = document.Pages.Count - 1; i >= 0; i--)
            {
                // Check if the current page is blank.
                if (document.Pages[i].IsBlank())
                {
                    // Remove the blank page from the document.
                    document.Pages.RemoveAt(i);
                }
                else
                {
                    // Convert the non-blank page to an image.
                    Image image = document.SaveAsImage(i, PdfImageType.Bitmap);

                    // Check if the image is blank.
                    if (IsImageBlank(image))
                    {
                        // If the image is blank, remove the corresponding PDF page.
                        document.Pages.RemoveAt(i);
                    }

                    //////////////////Use the following code for netstandard dlls/////////////////////////
                    /*
                    var image = document.SaveAsImage(i, Spire.Pdf.Graphics.PdfImageType.Bitmap);
                    image.Position = 0;
                    byte[] bytes = new byte[image.Length];
                    image.Read(bytes, 0, bytes.Length);
                    var bmp = SkiaSharp.SKBitmap.Decode(bytes);
                    var resized = bmp.Resize(new SkiaSharp.SKImageInfo(1, 1), SkiaSharp.SKSamplingOptions.Default);
                    var pixel = resized.GetPixel(0, 0);
                    if (pixel == null || (pixel.Red > 240 && pixel.Green > 240 && pixel.Blue > 240))
                    {
                        document.Pages.RemoveAt(i);
                    }
                    */
                }
            }

            // Define the output file name.
            String result = "RemoveBlankPages_out.pdf";

            // Save the modified document to the specified output file.
            document.SaveToFile(result);

            //Launch the Pdf file
            PDFDocumentViewer(result);
        }

        public static bool IsImageBlank(Image image)
        {
            // Convert the Image object to a Bitmap object for pixel manipulation.
            Bitmap bitmap = new Bitmap(image);

            // Iterate through each pixel in the image.
            for (int i = 0; i < bitmap.Width; i++)
            {
                for (int j = 0; j < bitmap.Height; j++)
                {
                    // Get the color of the current pixel.
                    Color pixel = bitmap.GetPixel(i, j);

                    // Check if any of the RGB values of the pixel are less than 240.
                    // If any of the RGB values are less than 240, it means the pixel is not blank.
                    if (pixel.R < 240 || pixel.G < 240 || pixel.B < 240)
                    {
                        // Return false since a non-blank pixel is found.
                        return false;
                    }
                }
            }

            // If all pixels are blank (i.e., all RGB values are greater than or equal to 240), return true.
            return true;
        }
        private void PDFDocumentViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}

File: VB/15_Document/RemoveBlankPages.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics

Namespace RemoveBlankPages
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new PdfDocument object
			Dim document As New PdfDocument()

			' Load the PDF document from the specified file path
			document.LoadFromFile("..\..\..\..\..\..\Data\RemoveBlankPages.pdf")

			' Iterate through the pages of the document in reverse order
			For i As Integer = document.Pages.Count - 1 To 0 Step -1

				' Check if the current page is blank
				If document.Pages(i).IsBlank() Then
					' Remove the blank page
					document.Pages.RemoveAt(i)
				Else
					' Convert the page to an image (bitmap)
					Dim image As Image = document.SaveAsImage(i, PdfImageType.Bitmap)

					' Determine whether the image is blank or not
					If IsImageBlank(image) Then
						' Delete the corresponding PDF page if the image is blank
						document.Pages.RemoveAt(i)
					End If
				End If

			Next i

			' Specify the result file path for saving the modified document
			Dim result As String = "RemoveBlankPages_out.pdf"

			' Save the modified document to the result file
			document.SaveToFile(result)

			' Close the PDF document
			document.Close()

			'Launch the Pdf file
			PDFDocumentViewer(result)
		End Sub

		Public Shared Function IsImageBlank(ByVal image As Image) As Boolean
			Dim bitmap As New Bitmap(image)
			For i As Integer = 0 To bitmap.Width - 1
				For j As Integer = 0 To bitmap.Height - 1
					Dim pixel As Color = bitmap.GetPixel(i, j)
					If pixel.R < 240 OrElse pixel.G < 240 OrElse pixel.B < 240 Then
						Return False
					End If
				Next j
			Next i
			Return True
		End Function

		Private Sub PDFDocumentViewer(ByVal filename As String)
			Try
				Process.Start(filename)
			Catch
			End Try
		End Sub
	End Class
End Namespace
